media/server/gstplayer/include/GstGenericPlayer.h

```diff
@@ -167,6 +168,8 @@ class GstGenericPlayer : public IGstGenericPlayer, public IGstGenericPlayerPriva
     GstElement *getDecoder(const MediaSourceType &mediaSourceType) override;
 
 private:
+    friend class ActualSink;
```

Comment:
if I see correctly getSinkChildIfAutoVideoSink/getSinkChildIfAutoAudioSink will be used in a new class only, so please remove that friend and move those methods to ActualSink class

Reply:
In general, we should avoid using "friend" keyword, because usually it means, that it's something wrong with class interface.